docs/plots/backends.rst

```diff
@@ -0,0 +1,362 @@
+  continuous quantity or an irregular grid, even if the current implementation
+  may not provide that flexibility.
+
+Annotations
+-----------
+
+Backends should also implement the following annotation functions. They do not
+depend on the data plotted, but just annotate the plot, e.g. a
+`~sherpa.plot.RatioPlot` shows the ratio between data and model and can use an
+annotation to mark the ``ratio=1`` line.
+
+- `~sherpa.plot.backend.BaseBackend.hline` (horizontal across the entire axes)
+- `~sherpa.plot.backend.BaseBackend.vline` (vertical across the entire axes)
+
+Other annotations (e.g. text labels) might be added to the API in the future.
+For this reason new backends should inherit from
+`~sherpa.plot.backend.BasicBackend`. Any function added to the API will be
+implemented in `~sherpa.plot.backend.BasicBackend` as a no-op with a
+warning to the user like "Feature XYZ is not available in your backend ". That
+way, all Sherpa plots can immediately make use of newly added functions without
+breaking existing plotting backends; the worst that happens is that not all
+annotation will be visible in every backend.
+
+Return values
+-------------
+
+Sherpa does not expect a specific return argument from any plotting function,
+but they are allowed to have return values if that is helpful for their internal
+implementation, e.g. in the matplotlib backend, plotting a line might return a
+line object so that error bars plotted later can use ``line.color`` to match the
+color of that line.
+
+Creating plots and panels, clearing and overplotting
+----------------------------------------------------
+At this stage, we keep the existing API for creating plot and panels, for
+clearing and overplotting, i.e. each of the plotting functions above accepts
+ the following arguments: title, xlabel, ylabel, xlog, ylog, overplot, clearwindow
+
+Multi-panels plot can be set with clear_window, set_subplot, set_jointplot
+[copy documentation from CAIO 4.14 pylab_backend.clear_window etc. into here because that describes use and function]
+
+Interaction with interactive plots in the UI
+--------------------------------------------
+Each backend has additional functions that are called before, during and after
+interactive plots (begin, exception, and end), and for the setup of 
+multi-panel plots [those are taken essentially unchanged from the 4.14 version, so can copy from code into specs]
+
+Other methods
+--------------
+
+Backends need to have a few methods
+
+- ``as_html_XXX`` (where XXX is a plot type) that are used for interactive
+  display in the notebook with ``_repr_html_``.  These functions take a plot
+  object and return an html representation as a string.
+- ``get_XXX_plot/hist_prefs`` (where XXX is a plot type) which returns a
+  dictionary of preferences that is used for displaying this plot.
+- `~sherpa.plot.backend.BasicBackend.get_latex_for_string` to format latex in strings.
```

Comment:
I am not proud of inventing `get_latex_for_string` so if there's any way to get rid of it I'd cheer!

Reply:
I think that's a fine method. Well defined purpose, easily tested, I like it!